### DISCUSSION TITLE
zh/guide/drivers/Crypt

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#252] [Announcements]
# zh/guide/drivers/Crypt

介绍 Crypt是什么？ 相当于一个两层密码的保险箱加密了，除了你自己有钥匙能打开别人谁也打不开，配置中的密码和盐值相当于钥匙 当然还有密码和盐值你自己也要保管好，如果丢失了你自己也打不开，除非你自己穷举或者猜出正确密码。 使用提醒 1. 如果你不了解想要使用本 加/解密 驱动请仔细查看一下每个文字 2. 请先在本地使用了解一下在搬到生产环境中部署使用...

https://alist.nn.ci/zh/guide/drivers/Crypt.html

**Comment by maxwell-lv** [comment]
Rclone Crypt 文档 链接的URL写错了

**Comment by at86397388** [comment]
1.这个crypt上传的文件还会不会被和谐呢？
2.选了文件夹和文件名的加密，上传下载性能有影响么？
3.设置好后，文件夹或文件名加密的方式还能改不？

**Comment by at86397388** [comment]
是不是所有都是全新上传，没有妙传？

发自我的iPhone


------------------ Original ------------------
From: ww ***@***.***&gt;
Date: Thu,Aug 10,2023 2:00 PM
To: alist-org/docs ***@***.***&gt;
Cc: at86397388 ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [alist-org/docs] zh/guide/drivers/Crypt (Discussion #252)





 
1,不会
 2.有一点但不多
 3.有备份随便改，没备份不建议

**Comment by at86397388** [comment]
为什么在Crypt驱动下，移动文件也感觉重新上传的样子，特别慢啊？

**Comment by at86397388** [comment]
感觉使用Crypt，对下载性能，或者和挂载 alist的性能影响还是蛮大的呢



发自我的iPhone


------------------ Original ------------------
From: ww ***@***.***&gt;
Date: Thu,Aug 10,2023 2:00 PM
To: alist-org/docs ***@***.***&gt;
Cc: at86397388 ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [alist-org/docs] zh/guide/drivers/Crypt (Discussion #252)





 
1,不会
 2.有一点但不多
 3.有备份随便改，没备份不建议

**Reply by RoundedKnot4342** [reply]
它要从服务器下载加密了的数据然后服务器解密之后传到你那的，肯定影响性能大www

**Comment by at86397388** [comment]
感觉crypt不能用了？上传图片可以，但大一点的视频，明明提示成功，但网盘里实际没有东西

**Comment by FoolBreader** [comment]
这个只能通过alist解密吗，我可不可以将加密内容下载下来自行解密

**Reply by theRainight** [reply]
可以的 需要使用rclone，在本地创建一个相同配置的加密目录，将加密文件放入即可解密

**Reply by mosder66** [reply]
请问加密软件是什么？加密软件就是rclone？

**Comment by wx2020** [comment]
解密速度太慢，无法做到实时看解密视频

**Comment by Jiang-lai** [comment]
版本v3.28.0，夸克网盘通过Crypt加密，下载小于100MB的文件可以满速，大于的话只有几百KB/s。

**Comment by sanliuyi901** [comment]
如果不从Crypt里下载文件，直接从网盘里下载加密过的文件，自己用什么方法才能解密打开？

**Reply by thesexday** [reply]
试一试把加密文件所在的本地硬盘挂在alist下（新增一个本机存储），然后新增一个crypt存储，设置加密后文件存储路径为你加密文件保存的本机存储路径，密码、加盐、还有其他加密方式跟你原加密网盘设的crypt保持一致，这样你新增的crypt存储里面应该能够读出下载的加密过的文件。我没有试过，不知行不行

**Comment by deng4304** [comment]
对于蓝奏云，加密文件会使后缀消失，而蓝奏云只允许上传特定后缀名，如果不加密文件名（只加后缀和加密文件本身），有些会提示存在非法字符，无法上传，包括一些存在版权的文件也不允许上传，可否在加密文件名的同时，保持统一后缀，如  .zip

**Comment by ypq123456789** [comment]
想问下缩略图选项-启用在缩略图文件夹下预先生成的缩略图，这个具体怎么使用？

**Comment by ypq123456789** [comment]
还有就是阿里加密的驱动，代理url不能用cf的workers，其他的onedrive、阿里云盘啥的都可以用代理url，为啥呢

**Comment by wy16W2pIilK1xgqN** [comment]
Crypt(加密) 
该模块能否直接实现本地硬盘目录和网盘的双向加密同步？
试用时发现，直接挂载 “本地存储” 目录到Crypt(加密)  的目录下，向该“本地存储” 目录添加文件，文件并不会被自动加密。

**Reply by Whos-JohnGalt** [reply]
虽然…但是，包装箱里的快递物品，不可能在同一个原子宇宙内同时存在两份拷贝。如果 Crypt 和 Rclone 的加密机制相同，就是一个进程对象；任何一个加密进程如果能形成副本，应该没有用户了吧😓

**Comment by ray5378** [comment]
实时解密在播放流媒体时几秒卡一下，完全没办法达到可用，能不能在服务器上设置个缓存空间，达到流畅播放的效果

**Reply by cancundeyingzi** [reply]
不至于吧5105解密都能跑50MB/s,瓶颈都在io了

**Comment by at86397388** [comment]
加密盘总是出现这个提示，然后就无法访问，需要到存储那里再保存一下就好了，为什么啊？
can't find remote storage: storage not found; rawPath:

**Comment by at86397388** [comment]
这个加密驱动的加密分级是啥？AES256么？
